frontend/src/modules/entitieslist/components/EntitiesList.tsx

```diff
@@ -197,29 +197,31 @@ export class EntitiesListBase extends React.Component<InternalProps> {
         entity: EntityType,
         replaceHistory?: boolean,
     ) => {
-        const { dispatch, router } = this.props;
+        const { dispatch, parameters, router, store } = this.props;
 
-        const state = this.props.store.getState();
+        const state = store.getState();
         const unsavedChangesExist = state[unsavedchanges.NAME].exist;
         const unsavedChangesIgnored = state[unsavedchanges.NAME].ignored;
 
-        dispatch(
-            unsavedchanges.actions.check(
-                unsavedChangesExist,
-                unsavedChangesIgnored,
-                () => {
-                    dispatch(batchactions.actions.resetSelection());
-                    dispatch(editor.actions.reset());
-                    dispatch(
-                        navigation.actions.updateEntity(
-                            router,
-                            entity.pk.toString(),
-                            replaceHistory,
-                        ),
-                    );
-                },
-            ),
-        );
+        if (entity.pk !== parameters.entity) {
```

Comment:
Can you invert the condition and return when true? Then you can restore the original (already pretty deep) indentation of the following block.